### ISSUE TITLE
bleadperl triggers new warnings in Regexp::Common

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#7079]
Migrated from [rt.perl.org#25655](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=25655) (status was 'resolved')

Searchable as RT25655$

**Comment by p5pRT** [comment]
### From david.dyck@fluke.com

#### Created by david.dyck@fluke.com



Recent bleading edge version have started causing
Regexp&#8203;::Common to start issuing warnings that they
didn't before\.

$ perl \-we 'use Regexp&#8203;::Common'
Variable "$s" is not available at \(re\_eval 47\) line 1\.
Variable "$x" is not available at \(re\_eval 47\) line 1\.

Does this indicate a bug in perl or in Regexp&#8203;::Common?
\(the error message makes it a bit difficult to
 identify the offending line\(s\) of code\.

make test with Regexp\-Common\-2\.113 passes all test\, but
it too issues many warnings during the tests\.

It looks like the warning is comming from
&nbsp;    Regexp\-Common\-2\.113/lib/Regexp/Common/comment\.pm

in code like&#8203;:

&nbsp;    sub \{
&nbsp;            use re 'eval';
&nbsp;            my \($s\, $x\);
&nbsp;            my $re = qr \{\\b\(\[A\-Za\-z\]\+\)\\b
&nbsp;                         \(?\(?\{\($s\, $x\) = \(0\, lc $^N\);
&nbsp;                              $s \+= $Regexp&#8203;::Common&#8203;::comment&#8203;::scores
&nbsp;                                    \[ord \(chop $x\) \- ord \('a'\)\] while length $x;
&nbsp;                              $s  >= 5 && $s \< 18\}\)XXX|\)\}x;
&nbsp;            $re;
&nbsp;        \};

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=low

Site configuration information for perl v5.9.0:

Configured by dcd at Wed Jan 28 16:15:20 PST 2004.

Summary of my perl5 (revision 5 version 9 subversion 0 patch 22244) configuration:
  Platform:
    osname=linux, osvers=2.4.25-pre6, archname=i686-linux
    uname='linux dd 2.4.25-pre6 #1 fri jan 16 13:23:13 pst 2004 i686 '
    config_args='-Accflags=-DPERL_DISABLE_PMC -Dmksymlinks -Dinstallusrbinperl -Uversiononly -Dusedevel -Doptimize=-O3 -g -de -Dcf_email=david.dyck@fluke.com'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef useithreads=undef usemultiplicity=undef
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-DPERL_DISABLE_PMC -DDEBUGGING -fno-strict-aliasing -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O3 -g',
    cppflags='-DPERL_DISABLE_PMC -DDEBUGGING -fno-strict-aliasing -I/usr/local/include'
    ccversion='', gccversion='egcs-2.91.66.1 19990314/Linux (egcs-1.1.2 release)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=4
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lgdbm -ldbm -ldb -ldl -lm -lc
    perllibs=-ldl -lm -lc
    libc=/lib/libc.so.5.4.44, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fpic', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
    DEVEL21538


@INC for perl v5.9.0:
    /usr/local/lib/perl5/5.9.0/i686-linux
    /usr/local/lib/perl5/5.9.0
    /usr/local/lib/perl5/site_perl/5.9.0/i686-linux
    /usr/local/lib/perl5/site_perl/5.9.0
    /usr/local/lib/perl5/site_perl
    .


Environment for perl v5.9.0:
    HOME=/home/dcd
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/dcd/bin:/sbin:/usr/local/bin:/bin:/usr/bin:/usr/X11/bin:/usr/games:/usr/local/samba:/home/hobbes/tools/scripts:/home/hobbes/tools/linux:/usr0/hobbes/tools/scripts:/usr0/dcd/bin:/apps/general/bin:/usr/public
    PERL5_CPANPLUS_CONFIG=/home/dcd/.cpanplus/config
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @iabyn

On Thu\, Jan 29\, 2004 at 07&#8203;:39&#8203;:59AM \-0000\, David Dyck wrote&#8203;:
> Recent bleading edge version have started causing
> Regexp&#8203;::Common to start issuing warnings that they
> didn't before\.
> 
> $ perl \-we 'use Regexp&#8203;::Common'
> Variable "$s" is not available at \(re\_eval 47\) line 1\.
> Variable "$x" is not available at \(re\_eval 47\) line 1\.
> 
> Does this indicate a bug in perl or in Regexp&#8203;::Common?
> \(the error message makes it a bit difficult to
>  identify the offending line\(s\) of code\.
> 
> make test with Regexp\-Common\-2\.113 passes all test\, but
> it too issues many warnings during the tests\.
> 
> It looks like the warning is comming from
>     Regexp\-Common\-2\.113/lib/Regexp/Common/comment\.pm
> 
> in code like&#8203;:
> 
>     sub \{
>             use re 'eval';
>             my \($s\, $x\);
>             my $re = qr \{\\b\(\[A\-Za\-z\]\+\)\\b
>                          \(?\(?\{\($s\, $x\) = \(0\, lc $^N\);
>                               $s \+= $Regexp&#8203;::Common&#8203;::comment&#8203;::scores
>                                     \[ord \(chop $x\) \- ord \('a'\)\] while length $x;
>                               $s  >= 5 && $s \< 18\}\)XXX|\)\}x;
>             $re;
>         \};

Well\, the use of lexical variables in conjunction with \(?\{\.\.\.\}\)
is currently highly buggy\. My recent closure patches to bleedperl
now tend to cause warnings where before things just silently did the wrong
thing\.

It's on my List Of Things To Do\, but probably not any time soon \(since the
regex engine scares the life out of me\)\.


\-\- 
But Pity stayed his hand\. "It's a pity I've run out of bullets"\,
he thought\. \- "Bored of the Rings"

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @jkeenan

On Sun Feb 01 19&#8203;:09&#8203;:41 2004\, davem@&#8203;fdisolutions\.com wrote&#8203;:
> On Thu\, Jan 29\, 2004 at 07&#8203;:39&#8203;:59AM \-0000\, David Dyck wrote&#8203;:
> > Recent bleading edge version have started causing
> > Regexp&#8203;::Common to start issuing warnings that they
> > didn't before\.
> >
> > $ perl \-we 'use Regexp&#8203;::Common'
> > Variable "$s" is not available at \(re\_eval 47\) line 1\.
> > Variable "$x" is not available at \(re\_eval 47\) line 1\.
> >
> > Does this indicate a bug in perl or in Regexp&#8203;::Common?
> > \(the error message makes it a bit difficult to
> >  identify the offending line\(s\) of code\.
> >
> > make test with Regexp\-Common\-2\.113 passes all test\, but
> > it too issues many warnings during the tests\.
> >
\[snip\]
> 
> Well\, the use of lexical variables in conjunction with \(?\{\.\.\.\}\)
> is currently highly buggy\. My recent closure patches to bleedperl
> now tend to cause warnings where before things just silently did the
> wrong
> thing\.
> 
> It's on my List Of Things To Do\, but probably not any time soon \(since
> the
> regex engine scares the life out of me\)\.
> 
> 

I just installed Regexp\-Common\-2011121001 on top of Perl 5\.14\.2
\(Linux/i386\)\, using the 'cpan' utility\.  I got no warnings during 'make
test'\.  Nor did I get warnings during 'prove \-v t/\*\.t'\.  Nor did I get
any warnings saying&#8203;:

&nbsp;    perl \-we 'use Regexp&#8203;::Common'

I suspect that the bug was fixed at some point in the last eight years
and recommend that the ticket be closed\.

Thank you very much\.
Jim Keenan

**Comment by p5pRT** [comment]
### From @cpansprout

On Tue Mar 27 18&#8203;:38&#8203;:01 2012\, jkeenan wrote&#8203;:
> On Sun Feb 01 19&#8203;:09&#8203;:41 2004\, davem@&#8203;fdisolutions\.com wrote&#8203;:
> > On Thu\, Jan 29\, 2004 at 07&#8203;:39&#8203;:59AM \-0000\, David Dyck wrote&#8203;:
> > > Recent bleading edge version have started causing
> > > Regexp&#8203;::Common to start issuing warnings that they
> > > didn't before\.
> > >
> > > $ perl \-we 'use Regexp&#8203;::Common'
> > > Variable "$s" is not available at \(re\_eval 47\) line 1\.
> > > Variable "$x" is not available at \(re\_eval 47\) line 1\.
> > >
> > > Does this indicate a bug in perl or in Regexp&#8203;::Common?
> > > \(the error message makes it a bit difficult to
> > >  identify the offending line\(s\) of code\.
> > >
> > > make test with Regexp\-Common\-2\.113 passes all test\, but
> > > it too issues many warnings during the tests\.
> > >
> \[snip\]
> > 
> > Well\, the use of lexical variables in conjunction with \(?\{\.\.\.\}\)
> > is currently highly buggy\. My recent closure patches to bleedperl
> > now tend to cause warnings where before things just silently did the
> > wrong
> > thing\.
> > 
> > It's on my List Of Things To Do\, but probably not any time soon \(since
> > the
> > regex engine scares the life out of me\)\.
> > 
> > 
> 
> I just installed Regexp\-Common\-2011121001 on top of Perl 5\.14\.2
> \(Linux/i386\)\, using the 'cpan' utility\.  I got no warnings during 'make
> test'\.  Nor did I get warnings during 'prove \-v t/\*\.t'\.  Nor did I get
> any warnings saying&#8203;:
> 
>     perl \-we 'use Regexp&#8203;::Common'
> 
> I suspect that the bug was fixed at some point in the last eight years
> and recommend that the ticket be closed\.

The underlying perl bug that the older Regexp&#8203;::Common triggered I
believe is still present\, but is part of what Dave Mitchell is currently
working on\.

\-\- 

Father Chrysostomos

**Comment by p5pRT** [comment]
### From @cpansprout

On Tue Mar 27 20&#8203;:46&#8203;:06 2012\, sprout wrote&#8203;:
> The underlying perl bug that the older Regexp&#8203;::Common triggered I
> believe is still present\, but is part of what Dave Mitchell is currently
> working on\.

And is now fixed in the commits leading up to eb58a7e122\.

\-\- 

Father Chrysostomos

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'open' to 'resolved'
